src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (!Object.is(this[i], null) && !Object.is(this[i], undefined)) {
```

Comment:
Consider using strict equality operator `===` instead of `Object.is()` for better readability and consistency. 🤓